### PR TITLE
feat: add Gemini-pro-1.5 to GeminiTextGenerator Tuning and Support score() method in Gemini-pro-1.5

### PR DESCRIPTION
**Pull request by shuoweil**
feat: add Gemini-pro-1.5 to GeminiTextGenerator Tuning and Support score() method in Gemini-pro-1.5

Bug: b/381936588 and b/344891364

### CODE REVIEW

File: bigframes/ml/llm.py [hunk 1]
```diff
@@ -892,8 +892,12 @@ def fit(
         Returns:
             GeminiTextGenerator: Fitted estimator.
         """
-        if self._bqml_model.model_name.startswith("gemini-1.5"):
-            raise NotImplementedError("Fit is not supported for gemini-1.5 model.")
+        # Support gemini-1.5 and gemini-pro
+        supported_models = ["gemini-pro", "gemini-1.5-pro-002", "gemini-1.5-flash-002"]
+        if self.model_name not in supported_models:
+            raise NotImplementedError(
+                "Score is not supported models other than gemini-pro or gemini-1.5 model."
```
Comment (GarrettWu) [inline bigframes/ml/llm.py:899]: be more explicit to "gemini-1.5-pro-002" and "gemini-1.5-flash-002".

Since other gemini 1.5 endpoints aren't supported.

Comment (GarrettWu) [inline bigframes/ml/llm.py:899]: This is in fit() in stead of score().

File: bigframes/ml/llm.py [hunk 2]
```diff
@@ -892,8 +892,12 @@ def fit(
...
+            )
```
Comment (GarrettWu) [inline bigframes/ml/llm.py:900]: I couldn't leave a comment at unchanged lines. Need to update line 905 to each endpoint respectively. (still use gemini-1.0-pro-002 for gemini-pro, but issuing a warning for that case maybe more appropriate)

File: bigframes/ml/llm.py [hunk 3]
```diff
@@ -1009,7 +1013,7 @@ def score(
             "text_generation", "classification", "summarization", "question_answering"
         ] = "text_generation",
     ) -> bpd.DataFrame:
-        """Calculate evaluation metrics of the model. Only "gemini-pro" model is supported for now.
+        """Calculate evaluation metrics of the model. Only "gemini-pro" and "gemini-1.5" models are supported for now.
```
Comment (GarrettWu) [inline bigframes/ml/llm.py:1016]: same, more explicit to "gemini-1.5-pro-002" and "gemini-1.5-flash-002"

File: bigframes/ml/llm.py [hunk 4]
```diff
@@ -1041,9 +1045,12 @@ def score(
         if not self._bqml_model:
             raise RuntimeError("A model must be fitted before score")
 
-        # TODO(ashleyxu): Support gemini-1.5 when the rollout is ready. b/344891364.
-        if self._bqml_model.model_name.startswith("gemini-1.5"):
-            raise NotImplementedError("Score is not supported for gemini-1.5 model.")
+        # Support gemini-1.5 and gemini-pro
+        supported_models = ["gemini-pro", "gemini-1.5-pro-002", "gemini-1.5-flash-002"]
```
Comment (GarrettWu) [inline bigframes/ml/llm.py:1049]: Move and share consts to top.

File: bigframes/ml/llm.py [hunk 5]
```diff
@@ -1041,9 +1045,12 @@ def score(
...
+        if self.model_name not in supported_models:
+            raise NotImplementedError(
+                "Score is not supported models other than gemini-pro or gemini-1.5 model."
```
Comment (GarrettWu) [inline bigframes/ml/llm.py:1052]: Same, more explicit. 

Also the sentence seems awkward.

File: tests/system/small/ml/test_llm.py [hunk 6]
```diff
@@ -413,9 +413,17 @@ def test_llm_palm_score_params(llm_fine_tune_df_default_index):
     )
 
 
-@pytest.mark.flaky(retries=2)
-def test_llm_gemini_pro_score(llm_fine_tune_df_default_index):
-    model = llm.GeminiTextGenerator(model_name="gemini-pro")
+# test score() function for "gemini-pro" and "gemini-1.5" model
```
Comment (GarrettWu) [inline tests/system/small/ml/test_llm.py:416]: nit: test name already indicates it. The comment seems redundant.

File: tests/system/small/ml/test_llm.py [hunk 7]
```diff
@@ -413,9 +413,17 @@ def test_llm_palm_score_params(llm_fine_tune_df_default_index):
...
+@pytest.mark.parametrize(
+    "model_name",
+    (
+        "gemini-pro",
+        "gemini-1.5-pro-002",
+        "gemini-1.5-flash-002",
+    ),
+)
+def test_llm_gemini_score(llm_fine_tune_df_default_index, model_name):
```
Comment (GarrettWu) [inline tests/system/small/ml/test_llm.py:425]: add for next test "test_llm_gemini_pro_score_params" as well

**Review by GarrettWu — COMMENTED** [review]
PR title has typos.

It should be adding new features instead of "update".

File: bigframes/ml/llm.py [hunk 8]
```diff
@@ -892,8 +892,12 @@ def fit(
         Returns:
             GeminiTextGenerator: Fitted estimator.
         """
-        if self._bqml_model.model_name.startswith("gemini-1.5"):
-            raise NotImplementedError("Fit is not supported for gemini-1.5 model.")
+        # Support gemini-1.5 and gemini-pro
```
[the remark this answers is not in the document]
Comment (shuoweil) [reply bigframes/ml/llm.py:911]: done

File: bigframes/ml/llm.py (hunk 1 above)
Comment (shuoweil) [reply bigframes/ml/llm.py:915]: done

Comment (shuoweil) [reply bigframes/ml/llm.py:915]: I am working on two bugs,  b/381936588 and b/344891364, both fit() and score() needed to be updated.

File: bigframes/ml/llm.py (hunk 3 above)
Comment (shuoweil) [reply bigframes/ml/llm.py:1032]: done

File: bigframes/ml/llm.py (hunk 5 above)
Comment (shuoweil) [reply bigframes/ml/llm.py:1068]: done

File: tests/system/small/ml/test_llm.py (hunk 6 above)
Comment (shuoweil) [reply tests/system/small/ml/test_llm.py:416]: done

File: tests/system/small/ml/test_llm.py (hunk 7 above)
Comment (shuoweil) [reply tests/system/small/ml/test_llm.py:425]: done

File: bigframes/ml/llm.py (hunk 2 above)
Comment (shuoweil) [reply bigframes/ml/llm.py:916]: done

File: bigframes/ml/llm.py (hunk 4 above)
Comment (GarrettWu) [reply bigframes/ml/llm.py:1067]: Not seeing the change? I mean add const variables to the top of the file like https://github.com/googleapis/python-bigquery-dataframes/blob/main/bigframes/ml/llm.py#L67

File: bigframes/ml/llm.py (hunk 1 above)
Comment (GarrettWu) [reply bigframes/ml/llm.py]: I mean the message is incorrect

File: bigframes/ml/llm.py (hunk 5 above)
Comment (GarrettWu) [reply bigframes/ml/llm.py]: The message is ill-formed. Maybe just "score() method only supports xxx models".

Comment (shuoweil) [reply bigframes/ml/llm.py]: done

File: bigframes/ml/llm.py (hunk 4 above)
Comment (shuoweil) [reply bigframes/ml/llm.py:1067]: done

File: bigframes/ml/llm.py (hunk 1 above)
Comment (shuoweil) [reply bigframes/ml/llm.py]: done

File: bigframes/ml/llm.py [hunk 9]
```diff
@@ -79,6 +79,16 @@
     _GEMINI_1P5_PRO_FLASH_PREVIEW_ENDPOINT,
     _GEMINI_2_FLASH_EXP_ENDPOINT,
 )
+_GEMINI_FINE_TUNE_ENDPOINTS = (
+    _GEMINI_PRO_ENDPOINT,
+    _GEMINI_1P5_PRO_002_ENDPOINT,
+    _GEMINI_1P5_FLASH_002_ENDPOINT,
+)
+_GEMINI_SCORE_ENDPOINTS = (
```
Comment (GarrettWu) [inline bigframes/ml/llm.py:87]: Can we merge the 2 lists? They are identical, don't think they will diverge.

**Review by GarrettWu — APPROVED** [review]

File: bigframes/ml/llm.py (hunk 9 above)
Comment (shuoweil) [reply bigframes/ml/llm.py]: done

**Pull request merged**